Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -342,18 +342,23 @@ void WebsiteDataStore::resolveDirectoriesIfNecessary()
         m_resolvedConfiguration->setCookieStorageFile(FileSystem::pathByAppendingComponent(resolvedCookieDirectory, FileSystem::pathFileName(m_configuration->cookieStorageFile())));
     }
 
-    // Do not back up cache type data.
-    std::array allCacheDirectories = {
-        resolvedApplicationCacheDirectory()
-        , resolvedMediaCacheDirectory()
-        , resolvedNetworkCacheDirectory()
-#if ENABLE(ARKIT_INLINE_PREVIEW)
-        , resolvedModelElementCacheDirectory()
-#endif
-    };
+    // These cache paths are not under Caches directory, so we need to explicitly
+    // exclude them from backup.
+    if (m_configuration->identifier()) {
```

Comment:
Why this new check?
Also, we might be creating directories for some of the configuration resolved path, is it something we should also do in a background thread?

Reply:
For this new check: for WebsiteDataStores created without identifier, the default paths of these types are under Caches/ or tmp/ directory, which are excluded by the system; for WebsiteDataStores created with identifier, default paths of all types are under WebsiteDataStore/[identifier]/ directory, so we need to excluded them explicitly.
Therefore we only need to exclude forWebsiteDataStores created with identifier.

I remember it's because sandbox extension needs to be created on existing paths, and we need to create sandbox extension for parameters.

Reply:
Can you add this comment in thé commit message?